workers/unity/Packages/io.improbable.gdk.testutils/Editor/SpatialdManager.cs

```diff
@@ -0,0 +1,199 @@
+                .RunAsync()
+                .ConfigureAwait(false);
+
+            if (result.ExitCode != 0)
+            {
+                throw new Exception($"Failed to start deployment with error: {string.Join("\n", result.Stderr)}");
+            }
+
+            var deploymentData = Json.Deserialize(string.Join("", result.Stdout));
+
+            var id = (string) ((Dictionary<string, object>) deploymentData["content"])["id"];
+
+            return new LocalDeployment(this, id, name, ProjectName);
+        }
+
+        public async Task StopLocalDeployment(LocalDeployment deployment)
+        {
+            var result = await RedirectedProcess.Command(SpotBinary)
+                .WithArgs("alpha", "deployment", "delete", "-i", deployment.Id)
+                .InDirectory(Common.SpatialProjectRootDir)
+                .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                .RunAsync()
+                .ConfigureAwait(false);
+
+            if (result.ExitCode != 0)
+            {
+                throw new Exception($"Failed to stop deployment with error: {string.Join("\n", result.Stderr)}");
+            }
+        }
+
+        public async Task<List<LocalDeployment>> GetRunningDeployments()
+        {
+            var result = await RedirectedProcess.Command(SpotBinary)
+                .WithArgs("alpha", "deployment", "list", "-p", ProjectName, "-f", "NOT_STOPPED_DEPLOYMENTS", "--json")
+                .InDirectory(Common.SpatialProjectRootDir)
+                .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                .RunAsync()
+                .ConfigureAwait(false);
+
+            var json = Json.Deserialize(string.Join("", result.Stdout));
+            var content = (Dictionary<string, object>) json["content"];
+
+            if (!content.TryGetValue("deployments", out var deploymentsObj))
+            {
+                return new List<LocalDeployment>();
+            }
+
+            var deploymentData = (List<object>) deploymentsObj;
+
+            return deploymentData
+                .OfType<Dictionary<string, object>>()
+                .Select(data =>
+                {
+                    var id = (string) data["id"];
+                    var name = (string) data["name"];
+                    return new LocalDeployment(this, id, name, ProjectName);
+                })
+                .ToList();
+        }
+
+        public void Dispose()
+        {
+            var result = RedirectedProcess.Command(Common.SpatialBinary)
+                .WithArgs("service", "stop")
+                .InDirectory(Common.SpatialProjectRootDir)
+                .RedirectOutputOptions(OutputRedirectBehaviour.None)
+                .RunAsync()
+                .Result;
+
+            if (result.ExitCode != 0)
+            {
+                Debug.LogWarning($"Failed to stop spatiald with error: {string.Join("\n", result.Stderr)}");
+            }
+        }
+
+        public class LocalDeployment : IDisposable
```

Comment:
Minor: could be a `struct`

Reply:
I did have it as so originally - but changed it, for a _reason_ (which I can't remember now)